File: 1125/02/snippet.cs
/*
public class ListNode
{
    public int val;
    public ListNode next;
    public ListNode (int x)
    {
        val = x;
    }
}*/
class Solution
{
    public ListNode ReverseList(ListNode pHead)
    {
        // write code here
        ListNode pre = null;
        ListNode cur = pHead;
        ListNode next = null;

        while(cur != null){
            next = cur.next;
            cur.next = pre;
            pre = cur;
            cur = next;
        }
        return pre;
    }
}

File: 1125/02/snippet.cpp
/*
struct ListNode {
	int val;
	struct ListNode *next;
	ListNode(int x) :
			val(x), next(NULL) {
	}
};*/
class Solution {
public:
    ListNode* ReverseList(ListNode* pHead) {
		if(pHead == NULL || pHead->next == NULL){
			return pHead;
		}
		ListNode* ans = ReverseList(pHead->next);
		pHead->next->next = pHead;
		pHead->next = NULL;
		return ans;
    }
};
